data-prepper-plugins/ml-inference-processor/src/main/java/org/opensearch/dataprepper/plugins/ml_inference/processor/client/S3ClientFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
I do not see any ML specific code here. Probably good idea to move it outside to a common directory. It can be done in a follow-up PR

Reply:
Cool, will add this in the next PR.